server/app/services/geo/esri/EsriClient.java

```diff
@@ -204,4 +222,86 @@ public CompletionStage<Optional<AddressSuggestionGroup>> getAddressSuggestions(A
               return Optional.of(addressCandidates);
             });
   }
+
+  /**
+   * Calls the external Esri query feature service layer to retrieve features of given address
+   * coordinates
+   *
+   * <p>@see <a
+   * href="https://developers.arcgis.com/rest/services-reference/enterprise/query-feature-service-layer-.htm">Query
+   * (Feature Service/Layer)</a>
+   *
+   * <p>Returns an empty optional under the following conditions:
+   *
+   * <ul>
+   *   <li>ESRI_ADDRESS_SERVICE_AREA_VALIDATION_VALUES or ESRI_ADDRESS_SERVICE_AREA_VALIDATION_URLS
+   *       is not configured.
+   *   <li>The external ESRI service returns an error.
+   *   <li>The external ESRI services returns a non 200 status.
+   * </ul>
+   *
+   * If the external ESRI serice returns an error or non 200 status, then the request is retried up
+   * to the configured ESRI_EXTERNAL_CALL_TRIES. If ESRI_EXTERNAL_CALL_TRIES is not configured, the
+   * tries default to 3.
+   */
+  @VisibleForTesting
+  CompletionStage<Optional<JsonNode>> fetchServiceAreaFeatures(
+      AddressLocation location, String validationUrl) {
+    WSRequest request = ws.url(validationUrl);
+    request.setContentType(ESRI_CONTENT_TYPE);
+    // "f" stands for "format", options are json and pjson (PrettyJson)
+    request.addQueryParameter("f", ESRI_RESPONSE_FORMAT);
+    request.addQueryParameter("geometryType", "esriGeometryPoint");
+    request.addQueryParameter("returnGeometry", "false");
+    request.addQueryParameter("outFields", "*");
+    String geo = "{'x':";
+    geo += location.getLongitude();
+    geo += ",'y':";
+    geo += location.getLatitude();
+    geo += ",'spatialReference':";
+    geo += location.getWellKnownId();
+    request.addQueryParameter("geometry", geo);
+
+    return tryRequest(request, this.ESRI_EXTERNAL_CALL_TRIES)
+        .thenApply(
+            res -> {
+              // return empty if still failing after retries
+              if (res.getStatus() != 200) {
+                return Optional.empty();
+              }
+              return Optional.of(res.getBody(json()));
+            });
+  }
+
+  /**
+   * Calls an external Esri service to get the service areas of the provided {@link
+   * AddressLocation}. Takes the returned service areas and returns a boolean specifying if the
+   * service area is included.
+   *
+   * @return an optional boolean if successful, or an empty optional if the request fails.
+   */
+  public CompletionStage<Optional<Boolean>> isAddressLocationInServiceArea(
+      String serviceArea, AddressLocation location) {
```

Comment:
**suggestion (blocking):** it's unclear what `serviceArea` is here. I'd guess it's the `value` or `label` of a service area. Passing in a string here feels like primitive obsession since we have a `EsriServiceAreaValidationOption` type, recommend passing that in instead.

Reply:
Great suggestion, updated!